internal/broker/broker.go

```diff
@@ -393,14 +394,42 @@ func (b *Broker) IsAuthenticated(sessionID, authenticationData string) (string,
 	if err = b.updateSession(sessionID, session); err != nil {
 		return AuthDenied, "", err
 	}
-	return access, data, nil
+
+	data, err := result.toJSON()
+	if err == nil {
+		return result.access, data, nil
+	}
+
+	slog.Debug(fmt.Sprintf("Impossible to convert to JSON %#v: %v", result, err))
+	if result.access == AuthDenied {
+		return result.access, data, nil
+	}
+	return AuthRetry, `{"message": "authentication data encoding failure"}`, nil
```

Comment:
Remember the happy path concept...

Reply:
I did, the reason why I went this way, it was to avoid nesting two times later, as initially I definitely errored earlier, but since that included further checks inside an if I went this way.